### PR TITLE
Adding Chuck Norris Endpoint Feature

### LINKED ISSUE #236: Chuck Norris Endpoint
**Is your feature request related to a problem? Please describe.**
No - it's an endpoint suggestion.  Hitting the end point returns a random Chuck Norris fact.

**Describe the solution you'd like**
A Chuck Norris Fact Endpoint

**Describe alternatives you've considered**
There is no alternative to Chuck Norris - this is common Knowledge

**Additional context**
Here is a picture of Chuck Norris 
image

### PR DESCRIPTION
**Pull request by computamike**
## Describe your changes

- Adding a Chuck Norris Fact end point.  
- Also added static file support to express, so that I could add a warning icon to the endpoint.  This offers the ability to limitedly link static files and add them into the mark up.  If this is not required, or not suitable - happy to remove. 

## Link to issue this resolves
Fixes #236

## Screenshot of changes(if relevant)
image

### CODE REVIEW

File: src/modules/chuck-norris/api/chuck-norris-routes.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ *         type: string
+ *         example: https://api.chucknorris.io/jokes/izjeqnjzteeqms8l8xgdhw
+ *         description: URL for this Fact on api.chucknorris.io
+ *       value:
+ *         type: string
+ *         example: Chuck Norris knows the last digit of pi.
+ */
+
+/**
+ * @openapi
```
Comment (ageddesi) [inline src/modules/chuck-norris/api/chuck-norris-routes.ts:37]: Hey, this seems to be a duplicate of the comment above :)

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (ageddesi) [inline .vscode/launch.json:1]: Can you add .vscode to the .gitignore file please

File: src/modules/chuck-norris/utils/scrape-facts.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        "BwMFn3FCQQaCSw0uEKkWLw"
+    ]
+    const FactsPerCateory = 200                // Number of records for each category too download
+    var RetryCount = 10;                        // Number of times to re-attempt an extract for a category.
+
+    // Arguments
+    const destination = process.argv[2];
+    const OutputFile = path.resolve(destination);   
+    
+    console.log("👊 Obtaining Chuck Norris facts")
```
Comment (ageddesi) [inline src/modules/chuck-norris/utils/scrape-facts.ts:53]: Do you mind removing the console.logs or wrap them in a .env file setting for knowing if you are in the dev env. Maybe we could make a generic logger as a ticket to look into that does this for us. So instead you would use something like 
mockedApi.log()

File: src/modules/chuck-norris/categories.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { InternetModule } from "@faker-js/faker"
+
+const facts = require("./api/chuckfacts.json")
+
+// const array = [{
```
Comment (ageddesi) [inline src/modules/chuck-norris/categories.ts:5]: Can we remove these parts of commented out code please :)

File: src/modules/chuck-norris/categories.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  fact => { 
+    fact.categories.map(
+      category =>{
+        if(!uniqueCatgories.has(category)){
+          uniqueCatgories.set(category,"")
+        }
+      })  
+    } )
+
+console.log(Array.from(uniqueCatgories.keys()))
```
Comment (ageddesi) [inline src/modules/chuck-norris/categories.ts:34]: Same rule on console.logs as mentioned before.

**Review by ageddesi — CHANGES_REQUESTED** [review]

File: src/modules/chuck-norris/categories.ts (hunk 4 above)
Comment (computamike) [reply src/modules/chuck-norris/categories.ts:5]: can do

File: src/modules/chuck-norris/utils/scrape-facts.ts (hunk 3 above)
Comment (computamike) [reply src/modules/chuck-norris/utils/scrape-facts.ts:53]: @ageddesi - This is the scraping script - this is called by issuing the command `npm chuck` - which then downloads a bunch of chuck norris facts.

File: src/modules/chuck-norris/api/chuck-norris-routes.ts (hunk 1 above)
Comment (computamike) [reply src/modules/chuck-norris/api/chuck-norris-routes.ts]: confirmed - have removed the duplicate

File: .vscode/launch.json (hunk 2 above)
Comment (computamike) [reply .vscode/launch.json]: have added .vscode to .gitignore

File: src/modules/chuck-norris/categories.ts (hunk 5 above)
Comment (computamike) [reply src/modules/chuck-norris/categories.ts:34]: this categories can go - it's not required (and was for testing my map method)

Comment (computamike) [reply src/modules/chuck-norris/categories.ts]: have removed the file - it was not required.

File: src/modules/chuck-norris/utils/scrape-facts.ts (hunk 3 above)
Comment (ageddesi) [reply src/modules/chuck-norris/utils/scrape-facts.ts:53]: Ahh ok cool. Thanks @computamike

**Review by ageddesi — DISMISSED** [review]

**Review by ageddesi — APPROVED** [review]

**Pull request merged**